branchff

```diff
@@ -92,7 +90,7 @@ gitlib::repo_state || common::exit 1
 ###############################################################################
 # MAIN
 ###############################################################################
-BASEDIR="/usr/local/google/$USER"
+BASEDIR="${TMPDIR}/${PROG}/$(date +%Y%m%d%H%M)"
```

Comment:
When I run the thing twice in one minute (because of some early error), I get `/tmp/branchff/201908160822/branchff-release-1.16 exists.  Delete? (y/N): `, otherwise it will just continue but with a new `WORKDIR`. No-one does the cleanup of the `WORKDIR`s that might build up over time.

I propose:
- get rid of `BASEDIR`, I don't think we need that
- set `TMPDIR` to something real random, like `$( mktemp -d )`
- set an `EXIT` trap which removes `TMPDIR`

Reply:
`TMPDIR` and `PROG` are defined in another script that calls this branchff? asking because of im not familiar with that  (and want learn 😄 )

Reply:
`TMPDIR` and `PROG` come from some of the sourced scripts/libs, most probably `./lib/common.sh` for those two cases.

Reply:
I like Hannes' suggestion

Reply:
@hoegaarden @tpepper -- I've updated the `BASEDIR` to `"${TMPDIR}/${PROG}/$(date +%Y%m%d%H%M%S)"` so runs within the same minute don't cause that error.

I like Hanne's suggestion as well, but I imagine bad things could happen (directories accidentally get removed) when a user decides to explicitly set `BASEDIR`, `WORKDIR`, or `TREE_ROOT` themselves and I'm not sure I want to introduce that possibility or have to code defensively against it.

We purposely leave the directory around, so that a Release Manager can inspect the commit history before or after push. Do we actually want to change that behavior?

On the whole, having something in `/tmp` isn't the worst thing.
My distro (Ubuntu 18.04) cleans up `/tmp` on boot and I believe RHEL variants have a cron that will do this.

Reply:
For `branchff`:
Neither `BASEDIR`, `WORKDIR` or `TREE_ROOT` are currently configurable by users. Only with your proposed change we are now using `TMPDIR`, which is configurable by the user. We should of course not remove any of the dirs we didn't create in the first place, regardless if they were set by the user or not. But if we'd only use temporary dirs (`$(mktemp -d)`) I don't see how we could unintentionally delete users' directories.
In general I just prefer tools that clean after themselves, and I never needed to inspect anything in `branchff`'s dir after the script ran to completion. YMMV & I won't block on that.

For `anago`:
I am not convinced that `${HOME}/anago` is the right place. But it is the default (on non-goobuntu machines) already anyway, `anago` should only be run on GCB, and there we set `BASEDIR` & `TMPDIR` explicitly anyway (like [so](https://github.com/kubernetes/release/blob/c32c094ff025d4d61fc27e575b489a174090c630/gcb/stage.yaml#L43-L44)). So i guess that's cool.